go/src/koding/db/mongodb/modelhelper/roles.go

```diff
@@ -130,3 +130,51 @@ func FetchAccountGroups(username string) ([]string, error) {
 
 	return slugList, nil
 }
+
+// FetchAccountGroups lists the groups of a given username
+func FetchAccountGroups(username string) (groups []*models.Group, err error) {
+	account, err := GetAccount(username)
+	if err != nil {
+		return nil, err
+	}
+
+	selector := Selector{
+		"sourceName": "JGroup",
+		"targetId":   account.Id,
+		"targetName": "JAccount",
+		"as":         bson.M{"$in": []string{"owner", "admin", "member"}},
+	}
+
+	rels, err := GetAllRelationships(selector)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(rels) == 0 {
+		return nil, nil
```

Comment:
I'm removing this entirely since `GetAllRelationships` function above will return `mgo.ErrNotFound` in case of empty result, so this check is redundant.